### PR TITLE
Logfile and rc.d for FreeBSD

### PR DESCRIPTION
**Pull request by Kernel-Error**
**Purpose of proposed changes**

Adds a Logfile and an FreeBSD rc.d script

### CODE REVIEW

File: postfix_mta_sts_resolver/utils.py [hunk 1]
```diff
@@ -18,10 +18,14 @@ def __contains__(self, e):
         return e in self.__members__
 
 
-def setup_logger(name, verbosity):
+def setup_logger(name, verbosity, logfile):
```
Comment (Snawoot) [inline postfix_mta_sts_resolver/utils.py]:
```suggestion
def setup_logger(name, verbosity, logfile=None):
```

File: postfix_mta_sts_resolver/utils.py [hunk 2]
```diff
@@ -18,10 +18,14 @@ def __contains__(self, e):
...
+
     logger = logging.getLogger(name)
     logger.setLevel(verbosity)
-    handler = logging.StreamHandler()
+    if logfile == "console":
```
Comment (Snawoot) [inline postfix_mta_sts_resolver/utils.py]:
```suggestion
    if logfile is None:
```

File: postfix_mta_sts_resolver/defaults.py [hunk 3]
```diff
@@ -3,5 +3,6 @@
 TIMEOUT = 4
 STRICT_TESTING = False
 CONFIG_LOCATION = "/etc/postfix/mta-sts-daemon.yml"
+LOGFILE = "/var/log/mta-sts.log"
```
Comment (Snawoot) [inline postfix_mta_sts_resolver/defaults.py]:
```suggestion
```

File: mta-sts-query [hunk 4]
```diff
@@ -27,7 +27,8 @@ def parse_args():
 
 def main():
     args = parse_args()
-    mainLogger = utils.setup_logger('MAIN', args.verbosity)
+    logtype = "console"
```
Comment (Snawoot) [inline mta-sts-query]:
```suggestion
```

File: mta-sts-query [hunk 5]
```diff
@@ -27,7 +27,8 @@ def parse_args():
...
+    mainLogger = utils.setup_logger('MAIN', args.verbosity, logtype)
```
Comment (Snawoot) [inline mta-sts-query]:
```suggestion
    mainLogger = utils.setup_logger('MAIN', args.verbosity)
```

File: mta-sts-daemon.yml.example [hunk 6]
```diff
@@ -1,5 +1,6 @@
 host: 127.0.0.1
 port: 8461
+logfile: /var/log/mta-sts.log
```
Comment (Snawoot) [inline mta-sts-daemon.yml.example]:
```suggestion
```

File: mta-sts-daemon [hunk 7]
```diff
@@ -38,6 +42,7 @@ def populate_cfg_defaults(cfg):
 
     cfg['host'] = cfg.get('host', defaults.HOST)
     cfg['port'] = cfg.get('port', defaults.PORT)
+    cfg['logfile'] = cfg.get('logfile', defaults.LOGFILE)
```
Comment (Snawoot) [inline mta-sts-daemon]:
```suggestion
```

Removal suggested since `cfg['logfile']` is not used anywhere

File: contrib/mta-sts-daemon [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+# FreeBSD rc.d script
+
+. /etc/rc.subr
+
+name="mta_sts_daemon"
+rcvar=`set_rcvar`
+
+load_rc_config $name
+mta_sts_daemon_enable=${mta_sts_daemon_enable:=NO}
+config="/usr/local/etc/postfix/mta-sts-daemon.yml"
```
Comment (Snawoot) [inline contrib/mta-sts-daemon]:
```suggestion
config="/usr/local/etc/postfix/mta-sts-daemon.yml"
log_file="/var/log/mta-sts.log"
```

File: contrib/mta-sts-daemon [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+mta_sts_daemon_user="mta-sts"
+
+command_interpreter="/usr/local/bin/python3.6"
+command="/usr/local/bin/mta-sts-daemon -c ${config}"
```
Comment (Snawoot) [inline contrib/mta-sts-daemon]:
```suggestion
command="/usr/local/bin/mta-sts-daemon -c ${config} -l ${log_file}"
```

File: mta-sts-daemon.yml.example (hunk 6 above)
Comment (Snawoot) [reply mta-sts-daemon.yml.example]: Removal suggested since cfg['logfile'] is not used anywhere

File: mta-sts-daemon [hunk 10]
```diff
@@ -25,6 +25,10 @@ def parse_args():
                         help="config file location",
                         metavar="FILE",
                         default=defaults.CONFIG_LOCATION)
+    parser.add_argument("-l", "--logfile",
+                        help="log file location",
+                        metavar="FILE",
```
Comment (Snawoot) [inline mta-sts-daemon]:
```suggestion
                        metavar="FILE")
```

File: mta-sts-daemon [hunk 11]
```diff
@@ -25,6 +25,10 @@ def parse_args():
...
+                        default=defaults.LOGFILE)
```
Comment (Snawoot) [inline mta-sts-daemon]:
```suggestion
```

File: README.md [hunk 12]
```diff
@@ -120,6 +120,8 @@ optional arguments:
   -c FILE, --config FILE
                         config file location (default: /etc/postfix/mta-sts-
                         daemon.yml)
+  -l FILE, --logfile FILE
+                        log file location (default: /var/log/mta-sts.log)
```
Comment (Snawoot) [inline README.md]:
```suggestion
                        log file location (default: None)                     
```

**Comment by Kernel-Error** [comment]
Looks good and I learned something.

**Comment by Snawoot** [comment]
@Kernel-Error Thanks!

**Pull request merged**